common/core/gpudevice.cpp

```diff
@@ -64,6 +59,10 @@ bool GpuDevice::Initialize() {
   HandleHWCSettings();
   InitializeHotPlugEvents();
 
+  if (!InitWorker()) {
```

Comment:
shouldn't be needed. DisplayManager is now created before DisplayManager is created.

Reply:
check with latest code.

Reply:
Yes, DisplayManager is created but not fully initialized. If InitWorker called just next to DisplayManager creation, thread GpuDevice will be put into running status. It has a chance to call methods of display manager but the later one might be not fully initialized yet. The consequence of it is HWC might be crash unexpectedly.

Reply:
in this case initializing thread doesn't make sense if lock_fd_ == -1 and also in HandleRoutine u can remove call to HandleHWCSettings

Reply:
Also u can remove initialization_state_lock_ in  InitializeHotPlugEvents and HandleHWCSettings

Reply:
For commit afa1e13, the GpuDevice thread initialization sequence works for me, thanks